src/borda_clj/client.clj

```diff
@@ -41,28 +44,37 @@
     (let [[v ch] (alts! [report-ch [measurements-ch measurements]])]
       (cond
         (= ch measurements-ch) (recur {})
-        (nil? v) (close! kill-ch)
+        (nil? v) (do (close! kill-ch) (>! measurements-ch measurements))
         :else (recur (apply collect measurements max-buffer-size v))))))
 
-(defn flush [measurements-ch global-dimensions send on-error]
+(defn flush [measurements-ch global-dimensions send on-error trace-metadata]
   (let [measurements (<!! measurements-ch)]
     (when (not-empty measurements)
-      (try
-        (send (merge-global-to-measurements global-dimensions (finalize-submit-counts measurements)))
```

Comment:
While doing this trace function I also noticed two problems with this line:

- exceptions while calling either `merge-global-to-measurements` (unlikely) and `finalize-submit-counts` (this has actually happened) will be swallowed by the enclosing `try` block and treated as problems sending data, which we are more likely to ignore, and

- We are reporting the unfinalized measurements without the global data, instead of the data we were actually trying to send.  The global data we would probably not miss, but I guess the finalized measurements are more useful? Both versions should be fine to resubmit.

Reply:
Looks like a good improvement.

Reply:
Yeah, marginally.  :)  The second bullet point (which was the initial motivation for me looking into this) should be ignored.